src/libraries/System.Text.Json/tests/Serialization/ReferenceHandlingTests.Deserialize.cs

```diff
@@ -0,0 +1,1264 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Text.Json.Serialization;
+using Xunit;
+
+namespace System.Text.Json.Tests
+{
+    public static partial class ReferenceHandlingTests
```

Comment:
Please add tests involving the `KeyValuePair<,>` type.

Reply:
It seems like deserializing into a KVP passing a payload containing `$id` metadata fails, I will need to validate that on the internal `JsonKeyValuePairConverter`. Thanks for bringing that up.

Reply:
Created https://github.com/dotnet/runtime/issues/1882 to track this.